File: scaffolds/functions/swift/Handler.swift
// {name} — Flux function (compiled to WASM via swiftwasm)
// Build: swiftc -target wasm32-unknown-wasi Handler.swift -o {name}.wasm

@_cdecl("{name}_handler")
func {Name}Handler(inputPtr: UnsafePointer<UInt8>, inputLen: UInt32) -> UInt64 {
// TODO: decode JSON at inputPtr
let response = #"{\"ok\":true}"#
let bytes    = Array(response.utf8)
let outPtr   = UnsafeMutablePointer<UInt8>.allocate(capacity: bytes.count)
outPtr.initialize(from: bytes, count: bytes.count)
return (UInt64(UInt(bitPattern: outPtr)) << 32) | UInt64(bytes.count)
}

File: scaffolds/functions/zig/handler.zig
// {name} — Flux function (compiled to WASM via zig build-lib)
// Build: zig build-lib handler.zig -target wasm32-freestanding -dynamic -o {name}.wasm
const std = @import("std");

var output_buf: [1024]u8 = undefined;

export fn {name}_handler(input_ptr: [*]const u8, input_len: u32) u64 {
    _ = input_ptr;
    _ = input_len;

    const resp = "{\"ok\":true}";
    @memcpy(output_buf[0..resp.len], resp);

    const ptr: u64 = @intFromPtr(&output_buf);
    return (ptr << 32) | resp.len;
}
